lib/tasks/compact_index.rake

```diff
@@ -1,18 +1,48 @@
 namespace :compact_index do
+  def yanked_at_time(version)
+    query = ["SELECT created_at FROM deletions,
+      (SELECT name, number FROM versions, rubygems WHERE versions.rubygem_id = rubygems.id
+        AND versions.id = ?) AS rv
+      WHERE deletions.number = rv.number AND deletions.rubygem = rv.name", version.id]
+    sanitize_sql = ActiveRecord::Base.send(:sanitize_sql_array, query)
+    pg_result = ActiveRecord::Base.connection.execute(sanitize_sql)
+    return pg_result.first['created_at'] if pg_result.first.present?
+    Time.now.utc
+  end
+
   desc "Fill yanked_at with current time"
-  task migrate_yanked_at: :environment do
-    Version.where(yanked_at: nil).find_in_batches do |versions|
-      versions.each { |v| v.update_attribute :yanked_at, Time.now.utc }
+  task backfill_yanked_at: :environment do
+    without_yanked_at = Version.where(indexed: false)
```

Comment:
@sonalkr132 Can you update this to also only look at Versions that don't have `yanked_at` set? (new PR)